forge-gui-mobile/src/forge/card/CardRenderer.java

```diff
@@ -874,355 +874,182 @@ public static void drawCardWithOverlays(Graphics g, CardView card, float x, floa
+                    iconCount += 1;
                 }
                 if (listHK.contains("B")) {
-                    CardFaceSymbols.drawSymbol("hexproofB", g, abiX, abiY, abiScale, abiScale);
-                    abiY += abiSpace;
-                    abiCount += 1;
+                    currentY = drawAbilityIcon(g, "hexproofB", currentX, currentY, iconScale, iconSpace, iconCount, cardX, cardY, cardWidth);
+                    iconCount += 1;
                 }
                 if (listHK.contains("U")) {
-                    CardFaceSymbols.drawSymbol("hexproofU", g, abiX, abiY, abiScale, abiScale);
-                    abiY += abiSpace;
-                    abiCount += 1;
+                    currentY = drawAbilityIcon(g, "hexproofU", currentX, currentY, iconScale, iconSpace, iconCount, cardX, cardY, cardWidth);
+                    iconCount += 1;
                 }
                 if (listHK.contains("G")) {
-                    CardFaceSymbols.drawSymbol("hexproofG", g, abiX, abiY, abiScale, abiScale);
-                    abiY += abiSpace;
-                    abiCount += 1;
+                    currentY = drawAbilityIcon(g, "hexproofG", currentX, currentY, iconScale, iconSpace, iconCount, cardX, cardY, cardWidth);
+                    iconCount += 1;
                 }
                 if (listHK.contains("W")) {
-                    CardFaceSymbols.drawSymbol("hexproofW", g, abiX, abiY, abiScale, abiScale);
-                    abiY += abiSpace;
-                    abiCount += 1;
+                    currentY = drawAbilityIcon(g, "hexproofW", currentX, currentY, iconScale, iconSpace, iconCount, cardX, cardY, cardWidth);
+                    iconCount += 1;
                 }
                 if (listHK.contains("monocolored")) {
-                    CardFaceSymbols.drawSymbol("hexproofC", g, abiX, abiY, abiScale, abiScale);
-                    abiY += abiSpace;
-                    abiCount += 1;
+                    currentY = drawAbilityIcon(g, "hexproofC", currentX, currentY, iconScale, iconSpace, iconCount, cardX, cardY, cardWidth);
+                    iconCount += 1;
                 }
             } else {
-                CardFaceSymbols.drawSymbol("hexproof", g, abiX, abiY, abiScale, abiScale);
-                abiY += abiSpace;
-                abiCount += 1;
+                currentY = drawAbilityIcon(g, "hexproof", currentX, currentY, iconScale, iconSpace, iconCount, cardX, cardY, cardWidth);
+                iconCount += 1;
             }
         } else if (card.getCurrentState().hasShroud()) {
-            if (abiCount > 5) {
-                abiY = cy + (abiSpace * (abiCount - 6));
-                abiX = cx + ((cw * 2) / 1.92f);
-            }
-            CardFaceSymbols.drawSymbol("shroud", g, abiX, abiY, abiScale, abiScale);
-            abiY += abiSpace;
-            abiCount += 1;
-        }
-        if (card.getCurrentState().hasVigilance()) {
-            if (abiCount > 5) {
```

Comment:
Well thanks for running your tool but sorry this doesn't seem trustworthy at all as you're already removing some parts of the logic here